### DISCUSSION TITLE
In CanDataFrame Message where are we setting the Payload value information

### DISCUSSION THREAD

**Discussion by ShivaName** [CoRE-RG/FiCo4OMNeT#14] [General]
packet CanDataFrame{
    string displayString;
	unsigned int canID;			// ID of the message
	bool rtr;			// true if remote-frame
	double period;			// Periodicy of the message
}

The above is the CanDataFrame ,on which field do **i need to set the Payload information.**

In initialFrameCreation("data", dataFrameIDsTokenizer, dataFramesPeriodicityTokenizer,
                dataLengthDataFramesTokenizer, initialDataFrameOffsetTokenizer); 

we are setting Payload_packet with the size but **where are we setting the Payload value?**
        cPacket *payload_packet = new cPacket;
        payload_packet->setTimestamp();
        payload_packet->setByteLength(dataFieldLength);
        can_msg->encapsulate(payload_packet);

![image](https://user-images.githubusercontent.com/109166971/178551140-9e87cfb4-06a1-4794-837d-c0adcef2f8d9.png)

I am refering to 8 bits Data or dataframe 
DATA– Used to store up to 64 data bits of application data to be transmitted.

**Comment by SaloM0nk** [comment]
Hello, 
I moved this issue to the discussions. 

In our model, the CANFrame itself does not have a field containing the payload length. We use OMNeT++ encapsulation to ecap the payload inside the CANFrame. The length of the payload packet is set before encapsulation and the the CANFrame size can be updated accordingly. This is done in the line "payload_packet->setByteLength(dataFieldLength);" which you posted. 
We only consider the size of the frames in our simulation, to verify timings in the network, and transfer no real data values. The value of the payload could be set by using a custom message type as payload_packet.

I hope this answers your questions otherwise feel free to reply in this discussion.

Regards
Timo
